src/formatted-codegen.js

```diff
@@ -542,14 +546,22 @@ export class ExtensibleCodeGen {
     return this.bracket(content, Sep.ARRAY_INITIAL, Sep.ARRAY_FINAL, Sep.ARRAY_EMPTY);
   }
 
-  reduceObjectAssignmentTarget(node, { properties }) {
-    let state = this.brace(this.commaSep(properties, Sep.OBJECT_BEFORE_COMMA, Sep.OBJECT_AFTER_COMMA), node, Sep.OBJECT_BRACE_INITIAL, Sep.OBJECT_BRACE_FINAL, Sep.OBJECT_EMPTY);
+  reduceObjectAssignmentTarget(node, { properties, rest }) {
+    let content = this.commaSep(properties, Sep.OBJECT_BEFORE_COMMA, Sep.OBJECT_AFTER_COMMA);
+    if (rest != null) {
+      content = seq(content, this.t(','), this.t('...'), this.sep(Sep.REST), rest);
```

Comment:
Strictly speaking this should `OBJECT_BEFORE_COMMA` and  `Sep.OBJECT_AFTER_COMMA` around the `,`. (Ditto below.)

I would very much like to get rid of the formatted codegen, or at least just hardcode a format rather than doing this "extensible" thing.